### PR TITLE
match functions in world partner files for lakilester, watt, sushie

### PR DESCRIPTION
**Pull request by Rainchus**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 13 functions (+0.11%) on `us`.

⚠️ This PR introduces 2 warnings:
- src/world/partner/watt.c:118: warning: implicit declaration of function `func_802BD1AC_31CD1C'
- src/world/partner/watt.c:99: warning: initialization makes pointer from integer without a cast

File: src/world/partner/sushie.c [hunk 1]
```diff
@@ -1,42 +1,150 @@
 #include "common.h"
+#include "../src/world/partners.h"
+#include "npc.h"
 
 extern unkPartnerStruct* D_802BFDF8_320B68;
+extern s32 D_802BFEEC;
+extern f32 D_802BFEE0;
+extern void func_802BD368_31E0D8(s32, f32, f32, f32, f32, f32);
+extern void playFX_23(s32, f32, f32, f32, s32);
```
Comment (ethteck) [inline src/world/partner/sushie.c:9]: if you include effects.h you won't need to declare this effect func. also please remove "extern" from func decls

File: src/world/partner/sushie.c [hunk 2]
```diff
@@ -1,42 +1,150 @@
...
+extern f32 D_802BFDB4_320B24;
+extern s16 D_8010C97A;
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BD100_31DE70);
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BD20C_31DF7C);
+void func_802BD100_31DE70(void) {
+    Npc* partnerNPC = get_npc_unsafe(NPC_PARTNER);
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* cam =  gCameras;
```
Comment (Brotenko) [inline src/world/partner/sushie.c:24]:
```suggestion
    Camera* cam = gCameras;
```

Comment (ethteck) [inline src/world/partner/sushie.c:24]: &gCameras[0] (minor gripe but it's more explicit that we're getting the first one)

File: src/world/partner/sushie.c [hunk 3]
```diff
@@ -1,42 +1,150 @@
...
+void func_802BD368_31E0D8(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, f32 arg5) {
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    f32 sp20;
+
+    arg2 += 100.0f;
+    sp20 = 200.0f;
+    
+    if (npc_raycast_down_ahead(arg0, &arg1, &arg2, &arg3, &sp20, arg4, arg5) == 0) {
+        collisionStatus->currentFloor = -1;
+        return;
```
Comment (ethteck) [inline src/world/partner/sushie.c:77]: else instead of return would be better imo

File: src/world/partner/sushie.c [hunk 4]
```diff
@@ -1,42 +1,150 @@
...
 
 INCLUDE_ASM(s32, "world/partner/sushie", func_802BD414_31E184);
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BE280_31EFF0);
+s32 func_802BE280_31EFF0(s32 arg0, f32* arg1, f32* arg2, f32* arg3, f32 arg4, f32 arg5, f32* arg6) {
+    f32 sp38, sp3C, sp40, sp44, sp48, sp4C, sp50, sp54, sp58;
+    s32 colliderRayResult;
+
+    sin_cos_rad((*arg6 * TAU) / 360.0f, &sp38, &sp3C);
+    sp3C = -sp3C;
```
Comment (Brotenko) [inline src/world/partner/sushie.c:91]: Do you actually have to do this assignment, or can you directly use `-sp3C` in the function call?

File: src/world/partner/sushie.c [hunk 5]
```diff
@@ -1,42 +1,150 @@
...
+    }
+
+    return colliderRayResult;
+}
 
 INCLUDE_ASM(s32, "world/partner/sushie", func_802BE3A4_31F114);
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BF520_320290);
+void func_802BF520_320290(Npc* npc) {
+    npc->collisionHeight = 0x18;
```
Comment (ethteck) [inline src/world/partner/sushie.c:105]: these nums in decimal since they're units of length

File: src/world/partner/sushie.c [hunk 6]
```diff
@@ -1,42 +1,150 @@
...
+    D_802BFEE4 = 0;
+    D_802BFEE8 = 0;
+    D_802BFEF0 = 0;
+    D_802BFEF4 = 0;
+}
+
+s32 func_802BF568_3202D8(Evt* script, s32 isInitialCall) {
+    Npc* npc = script->owner2.npc;
+ 
+    if (isInitialCall != 0) {
```
Comment (ethteck) [inline src/world/partner/sushie.c:118]: no != 0 since bool

File: src/world/partner/sushie.c [hunk 7]
```diff
@@ -1,42 +1,150 @@
...
+
+    return colliderRayResult;
+}
 
 INCLUDE_ASM(s32, "world/partner/sushie", func_802BE3A4_31F114);
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BF520_320290);
+void func_802BF520_320290(Npc* npc) {
+    npc->collisionHeight = 0x18;
+    npc->collisionRadius = 0x24;
```
Comment (Brotenko) [inline src/world/partner/sushie.c:106]: These should be decimals

File: src/world/partner/sushie.c [hunk 8]
```diff
@@ -97,15 +205,127 @@ s32 func_802BF5A0_320310(Evt* evt, s32 arg1) {
...
+        partner_clear_player_tracking(npc);
+    }
 }
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BF920_320690);
+s32 func_802BF964_3206D4(Evt* script, s32 isInitialCall) {
+    Npc* npc = script->owner2.npc;
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BF964_3206D4);
+    if (isInitialCall != 0) {
```
Comment (ethteck) [inline src/world/partner/sushie.c:224]: no != 0

Comment (Brotenko) [inline src/world/partner/sushie.c:224]:
```suggestion
    if (isInitialCall) {
```

File: src/world/partner/sushie.c [hunk 9]
```diff
@@ -97,15 +205,127 @@ s32 func_802BF5A0_320310(Evt* evt, s32 arg1) {
...
+        partner_init_put_away(npc);
+        gPlayerStatusPtr->animFlags &= 0xFFBFFFFF;
```
Comment (ethteck) [inline src/world/partner/sushie.c:226]: ~0x400000 (tip: do 0xFFFFFFFF - the value)

File: src/world/partner/sushie.c [hunk 10]
```diff
@@ -97,15 +205,127 @@ s32 func_802BF5A0_320310(Evt* evt, s32 arg1) {
...
+    }
+}
+
+s32 func_802BFAB8_320828(Evt* script, s32 isInitialCall) {
+    Npc* partnerNPC = get_npc_unsafe(NPC_PARTNER);
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    f32 temp_f0;
+    s32 funcTemp0;
+
+    if (isInitialCall != 0) {
```
Comment (ethteck) [inline src/world/partner/sushie.c:265]: no != 0

File: src/world/partner/sushie.c (hunk 9 above)
Comment (Brotenko) [inline src/world/partner/sushie.c:226]:
```suggestion
        gPlayerStatusPtr->animFlags &= ~0x00400000;
```

File: src/world/partner/sushie.c (hunk 10 above)
Comment (Brotenko) [inline src/world/partner/sushie.c:265]:
```suggestion
    if (isInitialCall) {
```

File: src/world/partner/sushie.c [hunk 11]
```diff
@@ -97,15 +205,127 @@ s32 func_802BF5A0_320310(Evt* evt, s32 arg1) {
...
+            partnerNPC->pos.y = D_802BFEE0 - (partnerNPC->collisionHeight * 0.5f);
+            temp_f0 = atan2(partnerNPC->pos.x, partnerNPC->pos.z, script->varTable[1], script->varTable[3]);
+            partnerNPC->currentAnim.w = 0x7000A;
+            partnerNPC->yaw = temp_f0;
+            partnerNPC->jumpScale = 0.0f;
+            partnerNPC->moveSpeed = 3.0f;
+            partnerNPC->moveToPos.x = partnerNPC->pos.x;
+            partnerNPC->moveToPos.y = partnerNPC->pos.y;
+            partnerNPC->moveToPos.z = partnerNPC->pos.z;
+            partnerNPC->flags = (partnerNPC->flags | 0x108) & ~0x200;
```
Comment (ethteck) [inline src/world/partner/sushie.c:290]: this would look better as 
```
partnerNPC->flags |= 0x100 | 0x8;
partnerNPC->flags &= ~0x200;
```

Comment (Brotenko) [inline src/world/partner/sushie.c:290]: Please use the NPC flags defined in enums.h

File: src/world/partner/watt.c [hunk 12]
```diff
@@ -74,12 +91,125 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
...
+ApiStatus func_802BD2B4_31CE24(Evt *script, s32 isInitialCall) {
+    PlayerData* playerData = &gPlayerData;
+    Entity* entity;
+    f32 sp10, sp14;
+    f32 new_var;
+    f32 temp_f0;
+    Npc* npc = script->owner2.npc;
+    
+    if (gPartnerActionStatus.actionState.b[1] == 0) {
+        if (isInitialCall != 0) {
```
Comment (ethteck) [inline src/world/partner/watt.c:103]: no != 0

File: src/world/partner/watt.c [hunk 13]
```diff
@@ -74,12 +91,125 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
...
+            }
+
+            if (D_802BE310 != 0) {
+                D_802BE310->unk_0C->unk_04 = npc->pos.x;
+                D_802BE310->unk_0C->unk_08 = npc->pos.y + 13.0f;
+                D_802BE310->unk_0C->unk_0C = npc->pos.z;
+            }
+
+            return 0;
+
```
Comment (ethteck) [inline src/world/partner/watt.c:135]: this empty line I think isn't necessary

File: src/world/partner/watt.c [hunk 14]
```diff
@@ -74,12 +91,125 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
...
+        } else {
```
Comment (ethteck) [inline src/world/partner/watt.c:136]: this doesn't need to be an else if we're returning right before it. I think I like the return, so you can remove this else

File: src/world/partner/sushie.c (hunk 9 above)
Comment (Brotenko) [reply src/world/partner/sushie.c:226]: Also take a look at enums.h and see if there are flags defined for this - If there isn't, then please setup a new enum for AnimationFlags

File: src/world/partner/watt.c [hunk 15]
```diff
@@ -74,12 +91,125 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
...
+                        D_802BE274_31DDE4->unk_14 += D_802BE2F0_31DE60;
+
+                        if (D_802BE274_31DDE4->unk_14 > 40.0f) {
+                            D_802BE274_31DDE4->unk_14 = 40.0f;
+                        }
+
+                        if (--D_802BE274_31DDE4->unk_00 == 0) {
+                            D_802BE274_31DDE4->unk_04++;
+                        }
+                        break;
```
Comment (ethteck) [inline src/world/partner/watt.c:181]: No empty line needed after break

File: src/world/partner/watt.c [hunk 16]
```diff
@@ -74,12 +91,125 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
...
+                        D_802BE274_31DDE4->unk_04++;
+                        break;
+                    case 3:
+                        partner_flying_update_player_tracking(npc);
+                        partner_flying_update_motion(npc);
+                        if (--D_802BE274_31DDE4->unk_00 == 0) {
+                            D_802BE274_31DDE4->unk_04 = 0;
+                            D_8010C954 = NULL;
+                        }
+
```
Comment (ethteck) [inline src/world/partner/watt.c:194]: this empty line isn't needed either

File: src/world/partner/watt.c [hunk 17]
```diff
@@ -74,12 +91,125 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
...
+                        break;
+                }
+            }
+
+        if (D_802BE310 != 0) {
+            D_802BE310->unk_0C->unk_04 = npc->pos.x;
+            D_802BE310->unk_0C->unk_08 = npc->pos.y + 13.0f;
+            D_802BE310->unk_0C->unk_0C = npc->pos.z;
+        }
+
```
Comment (ethteck) [inline src/world/partner/watt.c:204]: nor is this

File: src/world/partner/watt.c (hunk 12 above)
Comment (Brotenko) [inline src/world/partner/watt.c:103]:
```suggestion
        if (isInitialCall) {
```

File: src/world/partner/watt.c [hunk 18]
```diff
@@ -74,12 +91,125 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
...
+        return 0;
```
Comment (ethteck) [inline src/world/partner/watt.c:205]: This return 0 shouldn't be necessary since we fall into the next one immediately after

**Review by ethteck — CHANGES_REQUESTED** [review]
small thingz. good job!

File: src/world/partner/watt.c [hunk 19]
```diff
@@ -192,4 +322,48 @@ void func_802BE014_31DB84(void) {
...
+    PlayerStatus* playerStatus;
+    f32 temp;
+    f32 temp_f20;
+    f32 spriteFacingAngle;
+    s32 phi_v1;
+    
+    if (gPartnerActionStatus.actionState.b[0] != 0) {
+        spriteFacingAngle = gPlayerStatusPtr->spriteFacingAngle;
+        if ((spriteFacingAngle < 90.0f) || (spriteFacingAngle > 270.0f)) {
+            if (!(gPlayerStatusPtr->trueAnimation & 0x1000000)) {
```
Comment (Brotenko) [inline src/world/partner/watt.c:338]: This seems to be an animation flag (take a look at enums.h)

File: src/world/partner/watt.c [hunk 20]
```diff
@@ -192,4 +322,48 @@ void func_802BE014_31DB84(void) {
...
+                phi_v1 = -5;
+                temp = 0.6f;
+            } else {
+                phi_v1 = 5;
+                temp = 0.6f;
+            }
+        } else {
+            if (!(gPlayerStatusPtr->trueAnimation & 0x1000000)) {
```
Comment (Brotenko) [inline src/world/partner/watt.c:346]: enums.h

**Review by Brotenko — CHANGES_REQUESTED** [review]

File: src/world/partner/sushie.c (hunk 4 above)
Comment (Rainchus) [reply src/world/partner/sushie.c:91]: assignment seems to be required

**Review by Brotenko — APPROVED** [review]

**Review by ethteck — APPROVED** [review]

File: src/world/partner/sushie.c [hunk 21]
```diff
@@ -1,42 +1,149 @@
...
+    playerStatus->position.z = partnerNPC->pos.z;
+    playerStatus->targetYaw = partnerNPC->yaw;
+
+    if (playerStatus->spriteFacingAngle < 90.0f || playerStatus->spriteFacingAngle > 270.0f) {
+        phi_v1 = -8;
+    } else {
+        phi_v1 = 8;
+    }
+
+    playerStatus->position.z -= cos_rad(((((cam->currentYaw + playerStatus->spriteFacingAngle) - 90.0f) + phi_v1) * TAU) / 360.0f) * -4.0f;
```
Comment (nanaian) [inline src/world/partner/sushie.c:38]: Holy parentheses
```suggestion
    playerStatus->position.z -= cos_rad(cam->currentYaw + playerStatus->spriteFacingAngle - 90.0f + phi_v1) * TAU / 360.0f * -4.0f;
```

File: src/world/partner/sushie.c [hunk 22]
```diff
@@ -97,15 +204,128 @@ s32 func_802BF5A0_320310(Evt* evt, s32 arg1) {
...
             break;
     }
-    return 0;
+
+    return ApiStatus_BLOCK;
 }
 
-INCLUDE_ASM(s32, "world/partner/sushie", func_802BF920_320690);
+void func_802BF920_320690(Npc* npc) {
+    if (D_8010C954 != 0) {
```
Comment (nanaian) [inline src/world/partner/sushie.c:212]:
```suggestion
    if (D_8010C954 != NULL) {
```

File: src/world/partner/watt.c [hunk 23]
```diff
@@ -74,12 +90,120 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
...
+                    func_802BD1AC_31CD1C(1);
+                    npc->currentAnim.w = 0x60003;
+                }
+            } else if (D_802BE278_31DDE8 != 0) {
+                D_802BE278_31DDE8 = 0;
+                func_802BD1AC_31CD1C(0);
+                npc->currentAnim.w = 0x60001;
+            }
+
+            if (D_802BE310 != 0) {
```
Comment (nanaian) [inline src/world/partner/watt.c:127]:
```suggestion
            if (D_802BE310 != NULL) {
```

File: src/world/partner/watt.c [hunk 24]
```diff
@@ -74,12 +90,120 @@ ApiStatus func_802BD27C_31CDEC(Evt* script, s32 isInitialCall) {
...
+                npc->pos.z = (entity->position.z - (sp14 * D_802BE274_31DDE4->unk_0C));
+                D_802BE274_31DDE4->unk_10 = clamp_angle(D_802BE274_31DDE4->unk_10 - D_802BE274_31DDE4->unk_14);
+
+                if (D_802BE274_31DDE4->unk_0C > 20.0f) {
+                    D_802BE274_31DDE4->unk_0C--;
+                } else if (D_802BE274_31DDE4->unk_0C < 19.0f) {
+                    D_802BE274_31DDE4->unk_0C++;
+                }
+
+                temp_f0 = sin_rad((D_802BE274_31DDE4->unk_18 * TAU) / 360.0f);
```
Comment (nanaian) [inline src/world/partner/watt.c:159]:
```suggestion
                temp_f0 = sin_rad(D_802BE274_31DDE4->unk_18 * TAU / 360.0f);
```

File: src/world/partner/watt.c [hunk 25]
```diff
@@ -192,4 +316,48 @@ void func_802BE014_31DB84(void) {
...
+            if (!(gPlayerStatusPtr->trueAnimation & NPC_FLAG_1000000)) {
+                phi_v1 = 5;
+                temp = 0.6f;
+            } else {
+                phi_v1 = -5;
+                temp = 0.6f;
+            }
+        }
+
+        temp_f20 = ((((camera->currentYaw + 270.0f) - gPlayerStatusPtr->spriteFacingAngle) + phi_v1) * TAU) / 360.0f;
```
Comment (nanaian) [inline src/world/partner/watt.c:349]:
```suggestion
        temp_f20 = (camera->currentYaw + 270.0f - gPlayerStatusPtr->spriteFacingAngle + phi_v1) * TAU / 360.0f;
```

File: src/world/partner/watt.c [hunk 26]
```diff
@@ -192,4 +316,48 @@ void func_802BE014_31DB84(void) {
...
+
+        playerStatus = gPlayerStatusPtr;
+        partnerNPC = wPartnerNpc;
+        partnerNPC->pos.x = playerStatus->position.x + ((sin_rad(temp_f20) * gPlayerStatusPtr->colliderDiameter) * temp);
```
Comment (nanaian) [inline src/world/partner/watt.c:353]:
```suggestion
        partnerNPC->pos.x = playerStatus->position.x + (sin_rad(temp_f20) * gPlayerStatusPtr->colliderDiameter * temp);
```

File: src/world/partner/watt.c [hunk 27]
```diff
@@ -192,4 +316,48 @@ void func_802BE014_31DB84(void) {
...
+
+        new_var2 = wPartnerNpc;
+        playerStatus = gPlayerStatusPtr;
+        partnerNPC = new_var2;
+        partnerNPC->pos.z = playerStatus->position.z - ((cos_rad(temp_f20) * gPlayerStatusPtr->colliderDiameter) * temp);
```
Comment (nanaian) [inline src/world/partner/watt.c:358]:
```suggestion
        partnerNPC->pos.z = playerStatus->position.z - (cos_rad(temp_f20) * gPlayerStatusPtr->colliderDiameter * temp);
```

**Review by nanaian — CHANGES_REQUESTED** [review]

**Review by nanaian — APPROVED** [review]

**Pull request merged**